### PR TITLE
fix: readable error messages 

### PR DESCRIPTION
**Pull request by DBlanchard88**
- Mapped the error codes to easier to read error messages
- Created unit test to check functionality

 #1100

### CODE REVIEW

File: core/pkg/model/error.go [hunk 1]
```diff
@@ -8,3 +8,19 @@ const (
 	FlagDisabledErrorCode = "FLAG_DISABLED"
 	InvalidContextCode    = "INVALID_CONTEXT"
 )
+
+var readableErrorMessage = map[string]string{
+	FlagNotFoundErrorCode: "Flag not found",
+	ParseErrorCode:        "Error parsing input",
```
Comment (toddbaert) [inline core/pkg/model/error.go:14]:
```suggestion
	ParseErrorCode:        "Error parsing input or configuration",
```

**Review by toddbaert — APPROVED** [review]
Tested locally, works as expected. Thanks @DBlanchard88 , this has been open for a while!

Approved with minor suggestions.

**Review by toddbaert — APPROVED** [review]

**Review by Kavindu-Dodan — APPROVED** [review]
Nice improvement. I have this nit [1]

[1] - https://github.com/open-feature/flagd/pull/1325/files#r1644959707

File: core/pkg/model/error.go [hunk 2]
```diff
@@ -8,3 +8,19 @@ const (
...
+	GeneralErrorCode:      "General error",
+	FlagDisabledErrorCode: "Flag is disabled",
+	InvalidContextCode:    "Invalid context provided",
+}
+
+func GetErrorMessage(code string) string {
+	if msg, exists := ReadableErrorMessage[code]; exists {
+		return msg
+	}
+	return "An unknown error code"
```
[the remark this answers is not in the document]
Comment (DBlanchard88) [reply core/pkg/model/error.go]: That is a good point. I'll make the change.

**Pull request merged**